docs/supported-services/apigateway.rst

```diff
@@ -51,12 +56,38 @@ Parameters
      - No
      - false
      - If true, your Lambdas will be deployed inside your account's VPC.
+   * - custom_domains
+     - Array of :ref:`apigateway-custom-domains`
+     - No
+     -
+     - An array of custom domains to map to this API Gateway instance.
    * - tags
      - :ref:`tagging-resources`
      - No
      - 
      - Any tags you want to apply to your API Gateway app.
 
+.. _apigateway-custom-domains:
+
+Custom Domain Mappings
+~~~~~~~~~~~~~~~~~~~~~~
+.. NOTE::
+
+    This service does not currently support sharing custom domains between API Gateway instances using Base Path Mappings.
+    At this time, you can only map one API Gateway to one custom domain, with no path mapping.
+
+API Gateway allows for mapping gateways to one or more custom domains. These custom domains are always served via HTTPS.
+
+The Custom Domains section is defined by the following schema:
+
+.. code-block:: yaml
+
+    custom_domains:
+    - dns_name: <string> # The DNS name for the API Gateway. Must be a valid DNS name.
+      https_certificate: <arn> # The Amazon Certificate Manager certificate to use. This certificate must be in the us-east-1 region.
```

Comment:
This is an interesting comment about the certificate needing to be in the us-east-1 region, is this an AWS limitation? What happens if I want to use custom domains but am deploying to another region like us-west-2?

Reply:
It's a limitation of how they implement custom domains for API Gateway.  The dirty little secret is that API Gateway is essentially a fancy multi-tenant Cloudfront distribution. When you create a custom domain, it creates a new Cloudfront distribution under the hood. Cloudfront requires that all resources that it uses (certs, Lambda@Edge) be in us-east-1.

Lambda did recently add a new option - "Endpoint Configuration." The default value is 'edge', which matches the legacy way of using API Gateway and uses Cloudfront for everything. There's a new value, 'regional', which sets up a region-specific API endpoint that doesn't use Cloudfront and keeps the certs in the specific region. I decided that adding support for that was beyond the scope of this PR.  It wouldn't be hard, though, as we could basically just have a config like this:

```yml
  endpoint_configuration:
    types:
    - regional
    - edge
```

The two types of endpoint apparently aren't mutually exclusive.  We'd then have to add some extra logic to what I've added in this PR to handle that case and configure thing properly, but I don't think it'd be too hard.  The only real difficulty there would be that we need to have each `custom_domain` know which kind of endpoint it's supposed to point at.

Perhaps a better config for that would be to ditch the above and do this:

```yml
  custom_domains:
  - dns_name: regional.mycompany.com
     https_certificate: arn:acm:us-west-2:something
     type: 'regional'
  - dns_name: edge.mycompany.com
     https_certificate: arn:acm:us-east-1:something
     type: 'edge'
```
.
Then, based on the values in the custom domains, we set the values of 'EndpointConfiguration.Type', so the user doesn't have to worry about them.  We'd default to just 'Edge'.

FYI, here's the docs for the Cloudformation involved:

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-apigateway-restapi-endpointconfiguration.html